firstvoices/backend/tests/test_apis/base_api_test.py

```diff
@@ -357,6 +358,44 @@ def test_list_permissions(self):
             instance, site
         )
 
+    @pytest.mark.django_db
+    def test_create_assistant_permissions_valid(self):
+        site, user = factories.get_site_with_member(
+            site_visibility=Visibility.PUBLIC, user_role=Role.ASSISTANT
+        )
+
+        self.client.force_authenticate(user=user)
+
+        data = self.get_valid_data(site)
+        data["visibility"] = "team"
+
+        response = self.client.post(
+            self.get_list_endpoint(site_slug=site.slug),
+            data=self.format_upload_data(data),
+            content_type=self.content_type,
+        )
+
+        assert response.status_code == 201
+
+    @pytest.mark.django_db
+    def test_create_assistant_permissions_invalid(self):
+        site, user = factories.get_site_with_member(
+            site_visibility=Visibility.PUBLIC, user_role=Role.ASSISTANT
+        )
+
+        self.client.force_authenticate(user=user)
+
+        data = self.get_valid_data(site)
+        data["visibility"] = "public"
+
+        response = self.client.post(
+            self.get_list_endpoint(site_slug=site.slug),
+            data=self.format_upload_data(data),
+            content_type=self.content_type,
+        )
+
+        assert response.status_code == 400
```

Comment:
These are great test cases, but please move them to a mixin that is just for create APIs. Thanks! (You might already be doing this after our chat but I can't remember!)

Reply:
Done 😄